torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -551,23 +629,18 @@ struct GraphFuser {
       // TODO: Perhaps we should use cloneFrom now, as it seems unlikely
       // to copy select nodes now that we have refactored to have a Value
       // distinct from Node.
-      Node * input_chunk = block->owningGraph()->create(prim::ConstantChunk, 0);
-      input_chunk->addInput(input);
-      input_chunk->i_(attr::chunks, chunk->i(attr::chunks));
-      input_chunk->i_(attr::dim, chunk->i(attr::dim));
-      insertAt(&insertion_point, input_chunk);
-
+      bchunk->addInput(input);
       chunked_inputs.emplace_back(); // alas, to not be C++17
-      for (auto chunk_sel : chunk->outputs()) {
-          Value * input_chunk_sel = input_chunk->addOutput();
+      for (auto chunk_sel : producer_chunk_outputs) {
+          Value * input_chunk_sel = bchunk->addOutput();
           input_chunk_sel->setType(chunk_sel->type());
```

Comment:
FWIW this is not true if we were operating on `CompleteTensorType`s (it might get broadcast), but not sure how important is that

Reply:
Good point. I'll add an assert on the type for now

Reply:
Actually, I think this is valid for CompleteTensorType (do you have an example where it isn't?). All outputs of the `BroadcastingChunk` have the same shape; consider the following operations:
```
z = x * y
z0, z1 = BroadcastingChunk(z)
```
We only add new inputs to the BroadcastingChunk if they're inputs to nodes that are producers for the BroadcastingChunk. In this case, we can add `x`, `y` as inputs to the bchunk while removing `z`. The new outputs `x0, x1, y0, y1` will still have the same shape as `z0` and `z1` because the bchunk is equivalent to broadcasting `x, y` to be the same shape as `z` and then performing chunking.

Reply:
You're right!